collection/network/domain-trust.yml

```diff
@@ -0,0 +1,15 @@
+rule:
+  meta:
+    name: Detect Domain Trust Discovery 
+    namespace: collection/network
+    author: johnk3r
+    scope: function
+    att&ck:
+      - Discovery::Domain Trust Discovery  [T1482]
+    examples:
+      - 0796f1c1ea0a142fc1eb7109a44c86cb:0x40222F
+      - 0731679c5f99e8ee65d8b29a3cabfc6b:0x40408E
+  features:
+    - or:
+      - api: DsEnumerateDomainTrusts
+      - string: /nltest.exe/i
```

Comment:
How about removing the `.exe` to be a little more flexible?
Also should we add `DOMAIN_TRUSTS` as `nltest` can be used for various tasks?